### DISCUSSION TITLE
How to increase run time for cronjobs?

### DISCUSSION THREAD

**Discussion by gabrielchase** [vercel/community#4419] [Help]
### Summary

Within my `src/pages/api/crons/analytics/test.ts` I have this

```
import { prisma } from '../../../../server/db'
import { NextApiRequest, NextApiResponse } from 'next'

// https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration
export const config = {
  maxDuration: 300
}



export default async function handler(request: NextApiRequest, response: NextApiResponse) {
  console.log('now: ', new Date())
  console.log('waiting 30 seconds...')
  await new Promise((resolve) => setTimeout(resolve, 30000))

  console.log('waiting ANOTHER I 30 seconds...')
  await new Promise((resolve) => setTimeout(resolve, 30000))

  console.log('waiting ANOTHER II 30 seconds...')
  await new Promise((resolve) => setTimeout(resolve, 30000))

  console.log('waiting THE LAST 30 seconds...')
  await new Promise((resolve) => setTimeout(resolve, 30000))

  console.log('runtime final: ', new Date())
  response.status(200).json({
    success: true
  })
}

```

This is my `vercel.json`
```
{
  "crons": [
    {
      "path": "/api/crons/analytics/test",
      "schedule": "0 15 * * *"
    }
  ]
}
```

However, it times out after 15 seconds. **My organization is on the `Pro` tier.**

<img width="285" alt="Screenshot 2023-10-09 at 6 02 17 PM" src="https://github.com/vercel/community/assets/19620289/fd65ff1a-032d-4f45-b31e-c299d77721dd">

I'm following  this part of the documentation: https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration

<img width="662" alt="Screenshot 2023-10-09 at 6 10 10 PM" src="https://github.com/vercel/community/assets/19620289/e865170c-bd49-4a1c-8e09-77d0daa74320">

**Comment by amyegan** [comment]
Cron jobs [maxDuration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) limits are identical to [Serverless](https://vercel.com/docs/functions/serverless-functions#limits) and [Edge](https://vercel.com/docs/functions/edge-functions/limitations) function limits.

If you try to call that function directly (e.g. in browser, from you app, or via postman), do you also run into a 15s limit?

**Reply by gabrielchase** [reply]
<img width="269" alt="image" src="https://github.com/vercel/community/assets/19620289/01bf081d-135e-4d63-91d6-a1970d89aced">

Yup! When I manually call the API from the browser, I see in the vercel logs that it times out after 15 seconds.

> Cron jobs [maxDuration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) limits are identical to [Serverless](https://vercel.com/docs/functions/serverless-functions#limits) and [Edge](https://vercel.com/docs/functions/edge-functions/limitations) function limits.

Yeah that's why I'm confused on why this doesn't work. We've paid the plan + set the `config.maxDuration`

**Reply by z4nr34l** [reply]
@gabrielchase please check if same would happend if you migrate this endpoint to appdir and set

```tsx
export const maxDuration = 300;

// Method handler definition
```

In our projects it's working like should, but we are using appdir.

**Reply by amyegan** [reply]
What happens if you try changing the duration limit with a [`functions`](https://vercel.com/docs/projects/project-configuration#functions) object in your `vercel.json`?
```
"functions": {
  "api/crons/analytics/test.ts": {
    "maxDuration": 60 // This function can run for a maximum of 1 minute
  },
  "api/**/*.js": {
    "maxDuration": 30 // These functions can run for a maximum of 30 seconds
  }
}
```

**Reply by gabrielchase** [reply]
@amyegan I need it as a cronjob though. I want to schedule it everyday. It doesn't seem like serverless functions have a "schedule" property.

https://vercel.com/docs/functions/serverless-functions/quickstart

**Reply by gabrielchase** [reply]
> please check if same would happend if you migrate this endpoint to appdir and set

@z4nr34l  I actually also have it within the `appdir` (I assume this means, in the same place where the cronjob function is), but it doesn't seem to work either.

**Reply by cshaxu** [reply]
I run into exactly the same issue and I have the same "maxDuration" thing set to 60s but it still times out after 15s. Any ideas how to fix this?

**Reply by jeffroche** [reply]
Having a similar issue to the above but mine is an issue/question of precedence of setting the `maxDuration` config value. I have a value set in the functions object for all api endpoints but want to override one or more of the functions to have a longer `maxDuration` value. It seems like the functions object in `vercel.json` takes precedence over the exported config variable in the individual endpoint files though. Is that the case?

Specifically, if I have a `vercel.json` file with
```json
"functions": {
    "pages/api/**/*.ts": {
      "maxDuration": 60
    }
}
```

And then in my `api/cron/mycron.ts`, I have
```ts
export const config = {
  maxDuration: 180,
}
```
But I'm still seeing 60sec timeouts on `api/cron/mycron`. And am on the pro plan.

If we are setting the timeout using the functions object in `vercel.json` with a wildcard, is it not possible to override them in the individual files?
